Documentation/tutorial/howto_filter_out_unwanted_apis_attributes.md

```diff
@@ -0,0 +1,252 @@
+```
+  
+A complete **Sample** of the filter configuration file for filtering out APIs follows:
+
+```yaml
+apiRules:
+- exclude:
+    uidRegex: ^Microsoft\.TeamFoundation\.WorkItemTracking\.Proxy\.IRowSetsNative$
+- exclude:
+    uidRegex: ^Microsoft\.TeamFoundation\.WorkItemTracking\.Proxy\.MetadataRowSetsNative$
+- exclude:
+    uidRegex: ^Microsoft\.TeamFoundation\.WorkItemTracking\.Proxy\.RowSet\.Columns.*$
+    type: Member
+- exclude:
+    uidRegex: ^Microsoft\.TeamFoundation\.WorkItemTracking\.Proxy\.RowSetColumn\.Name.*$
+    type: Member
+- exclude:
+    hasAttribute:
+      uid: System.ComponentModel.EditorBrowsableAttribute
+      ctorArguments:
+      - System.ComponentModel.EditorBrowsableState.Never
+```
+
+### 2. Attribute Filter Rules
+
+To filter out Attributes, you could specify `attributeRules` with a list of `exclude` or `include` rules.
+
+The rules are similar to API filter. Please refer to [API Filter Rules](#1-api-filter-rules) section.
+
+### 3. Default Filter Configuration
+
+```yaml
+apiRules:
+- exclude:
+    hasAttribute:
+      uid: System.ComponentModel.EditorBrowsableAttribute
+      ctorArguments:
+      - System.ComponentModel.EditorBrowsableState.Never
+attributeRules:
+- exclude:
+    uidRegex: ^System\.ComponentModel\.Design$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.ComponentModel\.Design\.Serialization$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.Xml\.Serialization$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.Web\.Compilation$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.Runtime\.Versioning$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.Runtime\.ConstrainedExecution$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.EnterpriseServices$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.Diagnostics\.CodeAnalysis$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.Reflection\.DefaultMemberAttribute$
+    type: Type
+- exclude:
+    uidRegex: ^System\.CodeDom\.Compiler\.GeneratedCodeAttribute$
+    type: Type
+- include:
+    uidRegex: ^System\.ComponentModel\.(BindableAttribute|BrowsableAttribute|ComplexBindingPropertiesAttribute|DataObjectAttribute|DefaultBindingPropertyAttribute|ListBindableAttribute|LookupBindingPropertiesAttribute|SettingsBindableAttribute|TypeConverterAttribute)$
+    type: Type
+- include:
+    uidRegex: ^System\.ComponentModel\..+$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.ComponentModel\..+$
+    type: Type
+- include:
+    uidRegex: ^System\.Diagnostics\.(ConditionalAttribute|EventLogPermissionAttribute|PerformanceCounterPermissionAttribute)$
+    type: Type
+- include:
+    uidRegex: ^System\.Diagnostics\..+$
+    type: Namespace
+- exclude:
+    uidRegex: ^System\.Diagnostics\..+$
+    type: Type
+- include:
+    uidRegex: ^System\.Runtime\.CompilerServices\.ExtensionAttribute$
+    type: Type
+- include:
+    uidRegex: ^System\.Runtime\.CompilerServices\..+$
```

Comment:
current design to flatten namespace and type couldn't handle the scenario used in CAPS: to filter all types except XX in a namespace

---
In reply to: [78392603](https://github.com/dotnet/docfx/pull/686#discussion_r78392603) [](ancestors = 78392603)